app/src/main/java/com/jjswigut/matters/ui/MatterListRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,42 @@
+package com.jjswigut.matters.ui
+
+import android.graphics.ImageFormat
+import androidx.recyclerview.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import com.jjswigut.matters.R
+import com.jjswigut.matters.database.Matter
+
+
+
+class MatterListRecyclerViewAdapter(
+    //this started as a dummy list and kept it in to compile
+    val values: List<Matter>
+
+) : RecyclerView.Adapter<MatterListRecyclerViewAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.fragment_matter, parent, false)
+        return ViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val item = values[position]
+        holder.titleView.text = item.matterTitle
+        holder.contentView.text = item.matterContent
+    }
+
+    override fun getItemCount(): Int = values.size
+
+    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val titleView: TextView = view.findViewById(R.id.matter_title)
+        val contentView: TextView = view.findViewById(R.id.content)
+
+        override fun toString(): String {
```

Comment:
how come you're overriding toString?

Reply:
Another thing I didn't understand and got rid of.